pkg/infoschema/infoschema_v2.go

```diff
@@ -1043,6 +1043,14 @@ func loadTableInfo(ctx context.Context, r autoid.Requirement, infoData *Data, tb
 			))
 		}
 
+		// table is not public
+		if tblInfo.State != model.StatePublic {
```

Comment:
Does it means something wrong if we see the non-public tables?

Do you have any case to cover it? or this is just 'defensive programming'

Reply:
Maybe the only case is: user would see table being deleted from `information_schema.tables`.